src/assisted_installer_controller/assisted_installer_controller.go

```diff
@@ -1046,17 +1049,47 @@ func (c *controller) updateNodesLabels() bool {
 			continue
 		}
 		hostsWithLabels[hostname] = hostData
+		allNodeLabelsAsOneString += hostData.Host.NodeLabels
+	}
+	return hostsWithLabels, strings.Contains(allNodeLabelsAsOneString, roleLabel), nil
+}
+
+// Set labels to hosts
+// If one of the labels was role pause mcp and set this role
+// after setting all labels unpause in case it was paused
+func (c *controller) setNodesLabels() bool {
+	c.log.Infof("Setting node labels if require")
+	hostsWithLabels, pauseMCO, err := c.getAssistedHostsWithLabels()
```

Comment:
We should differentiate between labels on master nodes or worker nodes.

Reply:
Agree no need to pause mcp if master have new role, i believe it is not healthy to set custom roles on masters though

Reply:
Currently will check only workers in order to decide